### PR TITLE
timeline: edit/abort cleanups

### PR DESCRIPTION
**Pull request by bnjbvr**
A review commit by commit should be possible there. This makes the code less verbose, in preparation for sending reactions with the send queue, and reacting to local echoes.

### CODE REVIEW

File: crates/matrix-sdk-ui/src/timeline/event_item/mod.rs [hunk 1]
```diff
@@ -474,6 +485,24 @@ impl EventTimelineItem {
...
+        match &self.kind {
+            EventTimelineItemKind::Local(local) => {
+                if let Some(event_id) = local.event_id() {
+                    TimelineItemHandle::Remote(event_id)
+                } else {
+                    TimelineItemHandle::Local(
+                        local
+                            .send_handle
+                            .as_ref()
+                            .expect("EventTimelineItem::handle used in a unit test"),
```
Comment (andybalaam) [inline crates/matrix-sdk-ui/src/timeline/event_item/mod.rs:499]: This message is a bit confusing to me. Does it mean that we don't expect to find a missing `send_handle` except in a unit test? Can we guarantee that? Maybe something simpler like "Unexpected missing send_handle" would work?

File: crates/matrix-sdk-ui/src/timeline/mod.rs [hunk 2]
```diff
@@ -455,20 +456,22 @@ impl Timeline {
         })
     }
 
-    /// Given a transaction id, try to find a remote echo that used this
-    /// transaction id upon sending.
-    async fn find_remote_by_transaction_id(&self, txn_id: &TransactionId) -> Option<OwnedEventId> {
+    /// Returns a local or remote timeline item identified by this transaction
+    /// id.
+    async fn any_timeline_item_by_txn_id(
```
Comment (andybalaam) [inline crates/matrix-sdk-ui/src/timeline/mod.rs:461]: `find_item_by_transaction_id` might be a clearer name. The "any" part doesn't seem to add anything, although I understand it makes sense in the context that this method used to find only remote items.

**Review by andybalaam — APPROVED** [review]
Looks good! A couple of tiny suggestions.

File: crates/matrix-sdk-ui/src/timeline/event_item/mod.rs (hunk 1 above)
Comment (bnjbvr) [reply crates/matrix-sdk-ui/src/timeline/event_item/mod.rs:499]: > Does it mean that we don't expect to find a missing send_handle except in a unit test?
Yes

> Can we guarantee that?
We may try to sparkle generics there to mean presence or absence of the send handle, but it's overkill IMO.

> Unexpected missing send_handle
Sounds good!

File: crates/matrix-sdk-ui/src/timeline/mod.rs (hunk 2 above)
Comment (bnjbvr) [reply crates/matrix-sdk-ui/src/timeline/mod.rs:461]: There's already `item_by_transaction_id` which returns only local echoes, so I wanted something to distinguish it. But maybe the right move would be to rename `item_by_transaction_id` to something else :thinking:

Comment (andybalaam) [reply crates/matrix-sdk-ui/src/timeline/mod.rs:461]: Yeah, I would go for `local_item_by_transation_id` and `item_by_transaction_id` I guess.

**Pull request merged**